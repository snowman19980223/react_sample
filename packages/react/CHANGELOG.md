# @primer/react

## 36.8.0

### Minor Changes

- [#4214](https://github.com/primer/react/pull/4214) [`3dd498e6a7c28a1ec39b56912f76579f6674e022`](https://github.com/primer/react/commit/3dd498e6a7c28a1ec39b56912f76579f6674e022) Thanks [@langermank](https://github.com/langermank)! - Dialog component: add new `position` prop to support fullscreen and bottom sheets on narrow screens

- [#4236](https://github.com/primer/react/pull/4236) [`14a2776f18cf8f4a360b1887db64f8d5e904bc9d`](https://github.com/primer/react/commit/14a2776f18cf8f4a360b1887db64f8d5e904bc9d) Thanks [@joshblack](https://github.com/joshblack)! - Refactor project to avoid circular dependencies

### Patch Changes

- [#4199](https://github.com/primer/react/pull/4199) [`b6e58079a1517d2f1d31098b4e484b94625deabd`](https://github.com/primer/react/commit/b6e58079a1517d2f1d31098b4e484b94625deabd) Thanks [@siddharthkp](https://github.com/siddharthkp)! - experimental/SelectPanel: Improve keyboard navigation from search input

- [#4200](https://github.com/primer/react/pull/4200) [`dc988141216b91561a0ea0943b2cbff03be95029`](https://github.com/primer/react/commit/dc988141216b91561a0ea0943b2cbff03be95029) Thanks [@broccolinisoup](https://github.com/broccolinisoup)! - Tooltip v2: Allow external id to be passed down in tooltip so that the trigger can be labelled/described by the that id

- [#4246](https://github.com/primer/react/pull/4246) [`2aced1c61f26513ef2f7a5a4d05485c6fe7ec1ff`](https://github.com/primer/react/commit/2aced1c61f26513ef2f7a5a4d05485c6fe7ec1ff) Thanks [@siddharthkp](https://github.com/siddharthkp)! - Dialog v1: Fix font-family for title

- [#4265](https://github.com/primer/react/pull/4265) [`19c716f7f1ec44f1f1ee4a135b828208f2be058a`](https://github.com/primer/react/commit/19c716f7f1ec44f1f1ee4a135b828208f2be058a) Thanks [@siddharthkp](https://github.com/siddharthkp)! - SelectPanel2: Submit panel when an item is selected with `Enter` key

- [#4166](https://github.com/primer/react/pull/4166) [`c66b808cbefdbdd7482be752e5d311d2db4d23d8`](https://github.com/primer/react/commit/c66b808cbefdbdd7482be752e5d311d2db4d23d8) Thanks [@pksjce](https://github.com/pksjce)! - Fix focus styles in ActionList Item to support more than focus-visible

- [#4259](https://github.com/primer/react/pull/4259) [`c9fbef6438131cf358e2eaff3d866cf4398e622c`](https://github.com/primer/react/commit/c9fbef6438131cf358e2eaff3d866cf4398e622c) Thanks [@siddharthkp](https://github.com/siddharthkp)! - ActionList: Prevent scroll when an item is selected with `Space`

- [#4245](https://github.com/primer/react/pull/4245) [`d79a419116ea9e669c1ce598144a4b61ae15c52d`](https://github.com/primer/react/commit/d79a419116ea9e669c1ce598144a4b61ae15c52d) Thanks [@siddharthkp](https://github.com/siddharthkp)! - experimental/SelectPanel: Automatically adjust height based on contents

- [#4204](https://github.com/primer/react/pull/4204) [`da4469d9d958733e1b1dd32d6cd1a1bd0661ff91`](https://github.com/primer/react/commit/da4469d9d958733e1b1dd32d6cd1a1bd0661ff91) Thanks [@dylanatsmith](https://github.com/dylanatsmith)! - Increase Tooltip border-radius to match Primer View Components

- [#4205](https://github.com/primer/react/pull/4205) [`ff354604f70fc900cb79fe4c58eaee541a813d9f`](https://github.com/primer/react/commit/ff354604f70fc900cb79fe4c58eaee541a813d9f) Thanks [@pablonete](https://github.com/pablonete)! - Bugfix: Dialog.Header ignored the sx provided
