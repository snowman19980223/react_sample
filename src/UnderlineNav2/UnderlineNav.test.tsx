import React from 'react'
import '@testing-library/jest-dom/extend-expect'
import {fireEvent, render} from '@testing-library/react'
import {
  IconProps,
  CodeIcon,
  IssueOpenedIcon,
  GitPullRequestIcon,
  CommentDiscussionIcon,
  ProjectIcon,
  ShieldLockIcon,
  GraphIcon
} from '@primer/octicons-react'

import {UnderlineNav} from '.'

// window.matchMedia() is not implemented by JSDOM so we have to create a mock:
// https://jestjs.io/docs/manual-mocks#mocking-methods-which-are-not-implemented-in-jsdom
Object.defineProperty(window, 'matchMedia', {
  writable: true,
  value: jest.fn().mockImplementation(query => ({
    matches: false,
    media: query,
    onchange: null,
    addListener: jest.fn(), // deprecated
    removeListener: jest.fn(), // deprecated
    addEventListener: jest.fn(),
    removeEventListener: jest.fn(),
    dispatchEvent: jest.fn()
  }))
})

Object.defineProperty(window.Element.prototype, 'scrollTo', {
  value: jest.fn(),
  writable: true
})

const ResponsiveUnderlineNav = ({
  selectedItemText = 'Code',
  loadingCounters = false
}: {
  selectedItemText?: string
  loadingCounters?: boolean
}) => {
  const items: {navigation: string; icon?: React.FC<IconProps>; counter?: number}[] = [
    {navigation: 'Code', icon: CodeIcon},
    {navigation: 'Issues', icon: IssueOpenedIcon, counter: 120},
    {navigation: 'Pull Requests', icon: GitPullRequestIcon, counter: 13},
    {navigation: 'Discussions', icon: CommentDiscussionIcon, counter: 5},
    {navigation: 'Actions', counter: 4},
    {navigation: 'Projects', icon: ProjectIcon, counter: 9},
    {navigation: 'Insights', icon: GraphIcon},
    {navigation: 'Settings', counter: 10},
    {navigation: 'Security', icon: ShieldLockIcon}
  ]
  return (
    <UnderlineNav aria-label="Repository" loadingCounters={loadingCounters}>
      {items.map(item => (
        <UnderlineNav.Item
          key={item.navigation}
          icon={item.icon}
          selected={item.navigation === selectedItemText}
          counter={item.counter}
        >
          {item.navigation}
        </UnderlineNav.Item>
      ))}
    </UnderlineNav>
  )
}

describe('UnderlineNav', () => {
  it('renders aria-current attribute to be pages when an item is selected', () => {
    const {getByText} = render(<ResponsiveUnderlineNav />)
    const selectedNavLink = getByText('Code').closest('a')

    expect(selectedNavLink?.getAttribute('aria-current')).toBe('page')
  })
  it('renders aria-label attribute correctly', () => {
    const {container} = render(<ResponsiveUnderlineNav />)
    expect(container.getElementsByTagName('nav').length).toEqual(1)
    const nav = container.getElementsByTagName('nav')[0]

    expect(nav.getAttribute('aria-label')).toBe('Repository')
  })
  it('renders icons correctly', () => {
    const {container} = render(<ResponsiveUnderlineNav />)
    const nav = container.getElementsByTagName('nav')[0]
    expect(nav.getElementsByTagName('svg').length).toEqual(7)
  })
  it('fires onSelect on click and keypress', async () => {
    const onSelect = jest.fn()
    const {getByText} = render(
      <UnderlineNav aria-label="Test Navigation">
        <UnderlineNav.Item onSelect={onSelect}>Item 1</UnderlineNav.Item>
        <UnderlineNav.Item onSelect={onSelect}>Item 2</UnderlineNav.Item>
        <UnderlineNav.Item onSelect={onSelect}>Item 3</UnderlineNav.Item>
      </UnderlineNav>
    )
    const item = getByText('Item 1')
    fireEvent.click(item)
    expect(onSelect).toHaveBeenCalledTimes(1)
    fireEvent.keyPress(item, {key: 'Enter', code: 13, charCode: 13})
    expect(onSelect).toHaveBeenCalledTimes(2)
  })
  it('respects counter prop', () => {
    const {getByText} = render(<ResponsiveUnderlineNav />)
    const item = getByText('Issues').closest('a')
    const counter = item?.getElementsByTagName('span')[3]
    expect(counter?.className).toContain('CounterLabel')
    expect(counter?.textContent).toBe('120')
  })
  it('respects loadingCounters prop', () => {
    const {getByText} = render(<ResponsiveUnderlineNav loadingCounters={true} />)
    const item = getByText('Actions').closest('a')
    const loadingCounter = item?.getElementsByTagName('span')[2]
    expect(loadingCounter?.className).toContain('LoadingCounter')
    expect(loadingCounter?.textContent).toBe('')
  })
})
