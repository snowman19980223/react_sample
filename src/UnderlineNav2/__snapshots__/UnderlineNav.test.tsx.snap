// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UnderlineNav renders consistently 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  padding-left: 16px;
  padding-right: 16px;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  border-bottom: 1px solid;
  border-bottom-color: hsla(210,18%,87%,1);
  align: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  min-height: 48px;
}

.c3 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c5 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  margin-right: 8px;
}

.c6 {
  font-weight: 600;
}

.c8 {
  margin-left: 8px;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
}

.c9 {
  display: inline-block;
  padding: 2px 5px;
  font-size: 12px;
  font-weight: 600;
  line-height: 1;
  border-radius: 20px;
  background-color: rgba(175,184,193,0.2);
  color: #1F2328;
}

.c9:empty {
  display: none;
}

.c0 {
  position: absolute;
  width: 1px;
  height: 1px;
  padding: 0;
  margin: -1px;
  overflow: hidden;
  -webkit-clip: rect(0,0,0,0);
  clip: rect(0,0,0,0);
  white-space: nowrap;
  border-width: 0;
}

.c4 {
  color: #0969da;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #1F2328;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  line-height: calc(20/14);
  border-radius: 6px;
  font-size: 14px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: calc((2rem - 1.25rem) / 2);
  padding-bottom: calc((2rem - 1.25rem) / 2);
}

.c4:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c4:is(button) {
  display: inline-block;
  padding: 0;
  font-size: inherit;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c4 span[data-component="icon"] {
  color: #656d76;
}

.c4:focus {
  outline: 2px solid transparent;
}

.c4:focus {
  box-shadow: inset 0 0 0 2px #0969da;
}

.c4:focus:not(:focus-visible) {
  box-shadow: none;
}

.c4:focus-visible {
  outline: 2px solid transparent;
  box-shadow: inset 0 0 0 2px #0969da;
}

.c4 span[data-content]::before {
  content: attr(data-content);
  display: block;
  height: 0;
  font-weight: 600;
  visibility: hidden;
  white-space: nowrap;
}

.c4::after {
  position: absolute;
  right: 50%;
  bottom: calc(50% - 25px);
  width: 100%;
  height: 2px;
  content: "";
  background-color: #fd8c73;
  border-radius: 0;
  -webkit-transform: translate(50%,-50%);
  -ms-transform: translate(50%,-50%);
  transform: translate(50%,-50%);
}

.c7 {
  color: #0969da;
  -webkit-text-decoration: none;
  text-decoration: none;
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  color: #1F2328;
  text-align: center;
  -webkit-text-decoration: none;
  text-decoration: none;
  line-height: calc(20/14);
  border-radius: 6px;
  font-size: 14px;
  padding-left: 8px;
  padding-right: 8px;
  padding-top: calc((2rem - 1.25rem) / 2);
  padding-bottom: calc((2rem - 1.25rem) / 2);
}

.c7:hover {
  -webkit-text-decoration: underline;
  text-decoration: underline;
}

.c7:is(button) {
  display: inline-block;
  padding: 0;
  font-size: inherit;
  white-space: nowrap;
  cursor: pointer;
  -webkit-user-select: none;
  -moz-user-select: none;
  -ms-user-select: none;
  user-select: none;
  background-color: transparent;
  border: 0;
  -webkit-appearance: none;
  -moz-appearance: none;
  appearance: none;
}

.c7 span[data-component="icon"] {
  color: #656d76;
}

.c7:focus {
  outline: 2px solid transparent;
}

.c7:focus {
  box-shadow: inset 0 0 0 2px #0969da;
}

.c7:focus:not(:focus-visible) {
  box-shadow: none;
}

.c7:focus-visible {
  outline: 2px solid transparent;
  box-shadow: inset 0 0 0 2px #0969da;
}

.c7 span[data-content]::before {
  content: attr(data-content);
  display: block;
  height: 0;
  font-weight: 600;
  visibility: hidden;
  white-space: nowrap;
}

.c7::after {
  position: absolute;
  right: 50%;
  bottom: calc(50% - 25px);
  width: 100%;
  height: 2px;
  content: "";
  background-color: transparent;
  border-radius: 0;
  -webkit-transform: translate(50%,-50%);
  -ms-transform: translate(50%,-50%);
  transform: translate(50%,-50%);
}

.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  list-style: none;
  white-space: nowrap;
  padding-top: 0;
  padding-bottom: 0;
  padding-left: 0;
  padding-right: 0;
  margin: 0;
  margin-bottom: -1px;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  gap: 8px;
  position: relative;
}

@media (hover:hover) {
  .c4:hover {
    background-color: rgba(175,184,193,0.2);
    -webkit-transition: background .12s ease-out;
    transition: background .12s ease-out;
    -webkit-text-decoration: none;
    text-decoration: none;
  }
}

@media (forced-colors:active) {
  .c4::after {
    background-color: LinkText;
  }
}

@media (hover:hover) {
  .c7:hover {
    background-color: rgba(175,184,193,0.2);
    -webkit-transition: background .12s ease-out;
    transition: background .12s ease-out;
    -webkit-text-decoration: none;
    text-decoration: none;
  }
}

@media (forced-colors:active) {
  .c7::after {
    background-color: transparent;
  }
}

<div>
  <h2
    className="c0"
  >
    Repository navigation
  </h2>
  <nav
    aria-label="Repository"
    className="c1"
  >
    <ul
      className="c2"
      role="list"
    >
      <li
        className="c3"
      >
        <a
          aria-current="page"
          className="c4"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className="c5"
            data-component="icon"
          >
            <svg
              aria-hidden="true"
              className="octicon octicon-code"
              fill="currentColor"
              focusable="false"
              height={16}
              role="img"
              style={
                {
                  "display": "inline-block",
                  "overflow": "visible",
                  "userSelect": "none",
                  "verticalAlign": "text-bottom",
                }
              }
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="m11.28 3.22 4.25 4.25a.75.75 0 0 1 0 1.06l-4.25 4.25a.749.749 0 0 1-1.275-.326.749.749 0 0 1 .215-.734L13.94 8l-3.72-3.72a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215Zm-6.56 0a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042L2.06 8l3.72 3.72a.749.749 0 0 1-.326 1.275.749.749 0 0 1-.734-.215L.47 8.53a.75.75 0 0 1 0-1.06Z"
              />
            </svg>
          </span>
          <span
            className="c6"
            data-component="text"
            data-content="Code"
          >
            Code
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className="c5"
            data-component="icon"
          >
            <svg
              aria-hidden="true"
              className="octicon octicon-issue-opened"
              fill="currentColor"
              focusable="false"
              height={16}
              role="img"
              style={
                {
                  "display": "inline-block",
                  "overflow": "visible",
                  "userSelect": "none",
                  "verticalAlign": "text-bottom",
                }
              }
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="M8 9.5a1.5 1.5 0 1 0 0-3 1.5 1.5 0 0 0 0 3Z"
              />
              <path
                d="M8 0a8 8 0 1 1 0 16A8 8 0 0 1 8 0ZM1.5 8a6.5 6.5 0 1 0 13 0 6.5 6.5 0 0 0-13 0Z"
              />
            </svg>
          </span>
          <span
            className=""
            data-component="text"
            data-content="Issues"
          >
            Issues
          </span>
          <span
            className="c8"
            data-component="counter"
          >
            <span
              aria-hidden="true"
              className="c9"
            >
              120
            </span>
            <span
              className="c0"
            >
              Â (
              120
              )
            </span>
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className="c5"
            data-component="icon"
          >
            <svg
              aria-hidden="true"
              className="octicon octicon-git-pull-request"
              fill="currentColor"
              focusable="false"
              height={16}
              role="img"
              style={
                {
                  "display": "inline-block",
                  "overflow": "visible",
                  "userSelect": "none",
                  "verticalAlign": "text-bottom",
                }
              }
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="M1.5 3.25a2.25 2.25 0 1 1 3 2.122v5.256a2.251 2.251 0 1 1-1.5 0V5.372A2.25 2.25 0 0 1 1.5 3.25Zm5.677-.177L9.573.677A.25.25 0 0 1 10 .854V2.5h1A2.5 2.5 0 0 1 13.5 5v5.628a2.251 2.251 0 1 1-1.5 0V5a1 1 0 0 0-1-1h-1v1.646a.25.25 0 0 1-.427.177L7.177 3.427a.25.25 0 0 1 0-.354ZM3.75 2.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm0 9.5a.75.75 0 1 0 0 1.5.75.75 0 0 0 0-1.5Zm8.25.75a.75.75 0 1 0 1.5 0 .75.75 0 0 0-1.5 0Z"
              />
            </svg>
          </span>
          <span
            className=""
            data-component="text"
            data-content="Pull Requests"
          >
            Pull Requests
          </span>
          <span
            className="c8"
            data-component="counter"
          >
            <span
              aria-hidden="true"
              className="c9"
            >
              13
            </span>
            <span
              className="c0"
            >
              Â (
              13
              )
            </span>
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className="c5"
            data-component="icon"
          >
            <svg
              aria-hidden="true"
              className="octicon octicon-comment-discussion"
              fill="currentColor"
              focusable="false"
              height={16}
              role="img"
              style={
                {
                  "display": "inline-block",
                  "overflow": "visible",
                  "userSelect": "none",
                  "verticalAlign": "text-bottom",
                }
              }
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="M1.75 1h8.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 10.25 10H7.061l-2.574 2.573A1.458 1.458 0 0 1 2 11.543V10h-.25A1.75 1.75 0 0 1 0 8.25v-5.5C0 1.784.784 1 1.75 1ZM1.5 2.75v5.5c0 .138.112.25.25.25h1a.75.75 0 0 1 .75.75v2.19l2.72-2.72a.749.749 0 0 1 .53-.22h3.5a.25.25 0 0 0 .25-.25v-5.5a.25.25 0 0 0-.25-.25h-8.5a.25.25 0 0 0-.25.25Zm13 2a.25.25 0 0 0-.25-.25h-.5a.75.75 0 0 1 0-1.5h.5c.966 0 1.75.784 1.75 1.75v5.5A1.75 1.75 0 0 1 14.25 12H14v1.543a1.458 1.458 0 0 1-2.487 1.03L9.22 12.28a.749.749 0 0 1 .326-1.275.749.749 0 0 1 .734.215l2.22 2.22v-2.19a.75.75 0 0 1 .75-.75h1a.25.25 0 0 0 .25-.25Z"
              />
            </svg>
          </span>
          <span
            className=""
            data-component="text"
            data-content="Discussions"
          >
            Discussions
          </span>
          <span
            className="c8"
            data-component="counter"
          >
            <span
              aria-hidden="true"
              className="c9"
            >
              5
            </span>
            <span
              className="c0"
            >
              Â (
              5
              )
            </span>
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className=""
            data-component="text"
            data-content="Actions"
          >
            Actions
          </span>
          <span
            className="c8"
            data-component="counter"
          >
            <span
              aria-hidden="true"
              className="c9"
            >
              4
            </span>
            <span
              className="c0"
            >
              Â (
              4
              )
            </span>
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className="c5"
            data-component="icon"
          >
            <svg
              aria-hidden="true"
              className="octicon octicon-project"
              fill="currentColor"
              focusable="false"
              height={16}
              role="img"
              style={
                {
                  "display": "inline-block",
                  "overflow": "visible",
                  "userSelect": "none",
                  "verticalAlign": "text-bottom",
                }
              }
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="M1.75 0h12.5C15.216 0 16 .784 16 1.75v12.5A1.75 1.75 0 0 1 14.25 16H1.75A1.75 1.75 0 0 1 0 14.25V1.75C0 .784.784 0 1.75 0ZM1.5 1.75v12.5c0 .138.112.25.25.25h12.5a.25.25 0 0 0 .25-.25V1.75a.25.25 0 0 0-.25-.25H1.75a.25.25 0 0 0-.25.25ZM11.75 3a.75.75 0 0 1 .75.75v7.5a.75.75 0 0 1-1.5 0v-7.5a.75.75 0 0 1 .75-.75Zm-8.25.75a.75.75 0 0 1 1.5 0v5.5a.75.75 0 0 1-1.5 0ZM8 3a.75.75 0 0 1 .75.75v3.5a.75.75 0 0 1-1.5 0v-3.5A.75.75 0 0 1 8 3Z"
              />
            </svg>
          </span>
          <span
            className=""
            data-component="text"
            data-content="Projects"
          >
            Projects
          </span>
          <span
            className="c8"
            data-component="counter"
          >
            <span
              aria-hidden="true"
              className="c9"
            >
              9
            </span>
            <span
              className="c0"
            >
              Â (
              9
              )
            </span>
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className="c5"
            data-component="icon"
          >
            <svg
              aria-hidden="true"
              className="octicon octicon-graph"
              fill="currentColor"
              focusable="false"
              height={16}
              role="img"
              style={
                {
                  "display": "inline-block",
                  "overflow": "visible",
                  "userSelect": "none",
                  "verticalAlign": "text-bottom",
                }
              }
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="M1.5 1.75V13.5h13.75a.75.75 0 0 1 0 1.5H.75a.75.75 0 0 1-.75-.75V1.75a.75.75 0 0 1 1.5 0Zm14.28 2.53-5.25 5.25a.75.75 0 0 1-1.06 0L7 7.06 4.28 9.78a.751.751 0 0 1-1.042-.018.751.751 0 0 1-.018-1.042l3.25-3.25a.75.75 0 0 1 1.06 0L10 7.94l4.72-4.72a.751.751 0 0 1 1.042.018.751.751 0 0 1 .018 1.042Z"
              />
            </svg>
          </span>
          <span
            className=""
            data-component="text"
            data-content="Insights"
          >
            Insights
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className=""
            data-component="text"
            data-content="Settings"
          >
            Settings
          </span>
          <span
            className="c8"
            data-component="counter"
          >
            <span
              aria-hidden="true"
              className="c9"
            >
              10
            </span>
            <span
              className="c0"
            >
              Â (
              10
              )
            </span>
          </span>
        </a>
      </li>
      <li
        className="c3"
      >
        <a
          className="c7"
          href="#"
          onClick={[Function]}
          onKeyDown={[Function]}
        >
          <span
            className="c5"
            data-component="icon"
          >
            <svg
              aria-hidden="true"
              className="octicon octicon-shield-lock"
              fill="currentColor"
              focusable="false"
              height={16}
              role="img"
              style={
                {
                  "display": "inline-block",
                  "overflow": "visible",
                  "userSelect": "none",
                  "verticalAlign": "text-bottom",
                }
              }
              viewBox="0 0 16 16"
              width={16}
            >
              <path
                d="m8.533.133 5.25 1.68A1.75 1.75 0 0 1 15 3.48V7c0 1.566-.32 3.182-1.303 4.682-.983 1.498-2.585 2.813-5.032 3.855a1.697 1.697 0 0 1-1.33 0c-2.447-1.042-4.049-2.357-5.032-3.855C1.32 10.182 1 8.566 1 7V3.48a1.75 1.75 0 0 1 1.217-1.667l5.25-1.68a1.748 1.748 0 0 1 1.066 0Zm-.61 1.429.001.001-5.25 1.68a.251.251 0 0 0-.174.237V7c0 1.36.275 2.666 1.057 3.859.784 1.194 2.121 2.342 4.366 3.298a.196.196 0 0 0 .154 0c2.245-.957 3.582-2.103 4.366-3.297C13.225 9.666 13.5 8.358 13.5 7V3.48a.25.25 0 0 0-.174-.238l-5.25-1.68a.25.25 0 0 0-.153 0ZM9.5 6.5c0 .536-.286 1.032-.75 1.3v2.45a.75.75 0 0 1-1.5 0V7.8A1.5 1.5 0 1 1 9.5 6.5Z"
              />
            </svg>
          </span>
          <span
            className=""
            data-component="text"
            data-content="Security"
          >
            Security
          </span>
        </a>
      </li>
    </ul>
  </nav>
</div>
`;
